src/ts/main/src/pyodide-init.ts

```diff
@@ -0,0 +1,68 @@
+import {Action, Store} from "redux"
+import * as RootModule from "app/root"
+import {installPyodideStdIoActionAdapters} from "./install-pyodide-std-io-action-adapters"
+import {PROJECT_PYTHON_FILES} from "./managed-by-build"
+import {programRunAction} from "app/program-run"
+import {model} from "app/domain"
+import {writeFile} from "./pyodide-util"
+import {PyodidePythonExecutionEnvironment} from "./pyodide-python-execution-environment"
+
+declare const languagePluginLoader: Promise<any>
+declare const pyodide: any
+
+
+export function pyodideInit(start: number,
+                            store: Store<RootModule.RootState, Action>,
+                            onReadyCallback: (pythonEnv: model.PythonExecutionEnvironment) => void) {
+
+    // for more about languagePluginLoader, see https://github.com/iodide-project/pyodide/blob/master/src/pyodide.js
+    languagePluginLoader.then(() => {
+
+        installPyodideStdIoActionAdapters(window, pyodide, store.dispatch)
+
+        for (let relativePath in PROJECT_PYTHON_FILES) {
+            writeFile(pyodide._module.FS, "/.lib/" + relativePath, PROJECT_PYTHON_FILES[relativePath])
+        }
+
+        pyodide.runPython("print('hello world, from Python')")
+        pyodide.runPython(`import sys; sys.stderr.write("stderr this time - hello world, from Python\\n")`)
+
+        console.log(new Date().getTime() - start)
+
+        pyodide.runPython(`
+        
+        import ast
+        print(ast.parse("print('hello world, from Python AST')"))
+        
+        import sys
+        print("Python version")
+        print (sys.version)
+        print("Version info.")
+        print (sys.version_info)
+        
+        # make use of a function that comes from python code loaded is from the surrounding project
+        sys.path.append("/.lib/basic")
+        from append import append
+        print(append("appended hello ", "world"))
+        
+        `)
+
+        const python = new PyodidePythonExecutionEnvironment(pyodide)
+
+        python.runSingleModule({
```

Comment:
I know I balk at this a lot, but I don't like the differentiation between singleModule and multiModule - it's not a concept that is built into python, why do we need it? MultiModule should scale down to single module, but we shouldn't need a different mode. In other words, a single element list should not be a special case of a list.

Reply:
Got rid of this at the code level in a later PR (I was just getting started here).

I still find the product argument in favor of single-module special-casing to be attractive, but let's wait on hashing that out (because among other things there's the possibility that I'll come around to your POV in the meantime :) )